### ISSUE TITLE
How to use Pagination to print user ID of all people who have liked an image?

### ISSUE THREAD

**Issue by monajalal** [mgp25/Instagram-API#1024]
Basically what to do after this in order to print all the user_ids who have liked a specific image?
`$i->getMediaLikers(1462813261661883124);
`
The same question goes with, how to print the text of all the tags of a user?
`$i->getUserTags(53112486)
`
Thank you!

**Comment by mgp25** [comment]
You can use this example: https://github.com/mgp25/Instagram-API/blob/master/examples/PaginationExample.php

Make sure the response has that value, you can check it by doing a `var_dump` or checking debug's response

And using: `getUsernameId()`

Regards

**Comment by monajalal** [comment]
so `var_dump($likers)` gave this to me
http://pastebin.com/AWa5RVQ5
how can I only grab their pk? Is there a function you have defined for this? Also would this work in cases of 10k or so like? Is there a limit to this?

Thanks a lot!

**Comment by mgp25** [comment]
All vars can be obtained by:

`get` + `var`

in this case:

`getPk()`

Regards

**Comment by monajalal** [comment]
Sorry this sounds stupid but with no example on getting media likers or getting tags of a user this is hard to follow. I did both of these and got error:
```
 68     $i->getUserTags(53112486);
 69     foreach ($i as $liker){
 70         var_dump($liker.getPk());
 71     }
```
```
mona@pascal:~/computer_vision/instagram/Instagram-API/test$ php PaginationExample.php
PHP Fatal error:  Call to undefined function getPk() in 
/home/mona/computer_vision/instagram/Instagram-API/test/PaginationExample.php on line 70
mona@pascal:~/computer_vision/instagram/Instagram-API/test$ vi PaginationExample.php
 68     $i->getUserTags(53112486);
 69     foreach ($i as $liker){
 70         var_dump($liker.getUsername());
 71     }
mona@pascal:~/computer_vision/instagram/Instagram-API/test$ php PaginationExample.php
PHP Fatal error:  Call to undefined function getUsername() in /home/mona/computer_vision/instagram/Instagram-API/test/PaginationExample.php on line 70
```

For a user in mediaLikers we have:
`({"username": "h.pashazade", "pk": 1386547827, "full_name": "Hadi Pasha", "is_verified": false, "profile_pic_id": "1387607279436330599_1386547827", "is_private": true, "profile_pic_url": "http://scontent.cdninstagram.com/t51.2885-19/s150x150/14553221_206326719810937_2090433584522330112_a.jpg"},)`

Also
```
$ php PaginationExample.php
PHP Warning:  json_decode() expects parameter 1 to be string, object given in /home/mona/computer_vision/instagram/Instagram-API/test/PaginationExample.php on line 69
```
when I tried:
```
68     $i->getUserTags(53112486);
69     json_decode($i, true);
```

and 
$ php PaginationExample.php
PHP Fatal error:  Call to undefined function getResponse() in /home/mona/computer_vision/instagram/Instagram-API/test/PaginationExample.php on line 72
when I tried:
```
71 foreach ($i as $liker){
 72         var_dump($liker.getResponse().getUsers().getPk());
 73     }
```

Can up please provide the complete piece of code?

**Comment by mgp25** [comment]
@monajalal first of all, make sure you are using correct php sintaxis, this is not python or other language, you should do `->` instead of `.`

`$liker->getPk()`

Here is the PHP manual in case you need it: https://secure.php.net/manual/en/index.php

**Comment by monajalal** [comment]
```
 67     $i->getMediaLikers(1462813261661883124);
 68     foreach ($i as $liker){
 69         $liker->getPk();
 70     }
```

```
PHP Fatal error:  Call to a member function getPk() on string in /home/mona/computer_vision/instagram/Instagram-API/test/PaginationExample.php on line 69
mona@pascal:~/computer_vision/instagram/Instagram-API/test$ vi PaginationExample.php
```

**Comment by mgp25** [comment]
@monajalal i recommend you to see php syntax and code, because that doesnt make any sense...

Look at your foreach loop, what are you iteraring? `$i`?

**Comment by monajalal** [comment]
answer:
```
68      $likers = $i->getMediaLikers(1462813261661883124);
 69     #die(var_dump($likers)) ;
 70     foreach ($likers->users as $user) {
 71         //echo 'PK is '.$user->pk ."\n";
 72     }
```

**Comment by SteveJobzniak** [comment]
We're here for implementing the Instagram API. Not for customer support or PHP coding help. Because this is a free library and we're coding it as a hobby.
